src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -414,6 +455,15 @@ public Response get(String json) throws Exception
 
 			for (QueryMetric query : queries)
 			{
+				Map<String, Collection<String>> tags = query.getTags().asMap();
+				Set<String> keys = tags.keySet();
+
+				for (String key : keys) {
+					Iterator<String> iter = tags.get(key).iterator();
+					while(iter.hasNext())
+						span.setTag(key, iter.next());
+				}
+
 				queryMeasurementProvider.measureSpanForMetric(query);
 				queryMeasurementProvider.measureDistanceForMetric(query);
```

Comment:
I feel we are missing a bunch of good tags here. For instance they can tell us if the query failed/slow due to long distance or something else. May be we can have something like ``query_span_distance: hours|days|weeks|months`` for easier search as well? + we can log the query distance.

Reply:
We also have interesting log messages we can add for queries failures: 
``ueryMeasurementProvider.measureDistanceError(query);``